packages/apiserver/src/file/file.service.ts

```diff
@@ -1,65 +1,77 @@
-      }
-      fs.writeFileSync(path.resolve(filePath, fileName), file.buffer);
-      return `${process.env['NX_API_ROOT_URL']}/tmp/${fileName}`;
+      return new Promise((resolve, reject) => {
+        this.s3.putObject(
+          {
+            Bucket: 'mimir-content',
+            Key: fileName,
+            Body: file.buffer,
+            ACL: 'public-read',
+          },
+          (error: AWS.AWSError) => {
+            if (!error) {
+              resolve(
+                `${process.env['NX_API_SPACES']}/bookPictures/${fileName}`
+              );
+            } else {
+              reject(
+                new Error(
+                  `SpacesService_ERROR: ${
+                    error.message || 'Something went wrong'
+                  }`
+                )
+              );
+            }
+          }
+        );
+      });
+      return fileExtension;
     } catch (e) {
       throw new HttpException(e.message, HttpStatus.INTERNAL_SERVER_ERROR);
     }
   }
 
   async removeFile(fileName: string) {
     try {
-      fs.unlinkSync(path.resolve(process.cwd(), 'storage', 'tmp', fileName));
-      return 'File was deleted';
-    } catch (e) {
-      throw new HttpException(e.message, HttpStatus.INTERNAL_SERVER_ERROR);
-    }
-  }
-
-  moveFileInMainStorage(fileName: string | null, identifier: string) {
-    if (!fileName) return null;
-    if (/http/.test(fileName.split('//')[0])) return fileName;
-    try {
-      const onlyFileName = fileName.split('/').pop();
-      const filePathToStorage = path.resolve(
-        process.cwd(),
-        'storage',
-        'mainData'
-      );
-      const fileExtension = fileName.split('.').pop();
-      if (!fs.existsSync(filePathToStorage)) {
-        fs.mkdirSync(filePathToStorage, { recursive: true });
-      }
-      fs.copyFileSync(
-        path.resolve(process.cwd(), 'storage', 'tmp', onlyFileName),
-        path.resolve(process.cwd(), 'storage', 'mainData', onlyFileName)
-      );
-      fs.renameSync(
-        path.resolve(process.cwd(), 'storage', 'mainData', onlyFileName),
-        path.resolve(
-          process.cwd(),
-          'storage',
-          'mainData',
-          `${identifier}.${fileExtension}`
-        )
-      );
-      fs.unlinkSync(
-        path.resolve(process.cwd(), 'storage', 'tmp', onlyFileName)
-      );
-      return `${process.env['NX_API_ROOT_URL']}/mainData/${identifier}.${fileExtension}`;
+      return new Promise((resolve, reject) => {
+        this.s3.deleteObject(
+          {
+            Bucket: 'mimir-content',
+            Key: fileName,
+          },
+          (error: AWS.AWSError) => {
+            if (!error) {
+              resolve(`successfully deleted`);
+            } else {
+              reject(
+                new Error(
+                  `SpacesService_ERROR: ${
+                    error.message || 'Something went wrong'
+                  }`
+                )
+              );
+            }
+          }
+        );
+      });
```

Comment:
Don't think we will actually use it. Some random unused pictures are not concern for us

Reply:
wouldnt there be a waste of memory space, plus i noticed that this functionality is used like 3 times in frontend so i thought better to add. But i can delete it